### PR TITLE
Support listen() for unix sockets

### PR DESCRIPTION
**Pull request by stevenengler**
Added support for the listen() syscall, updated all of the existing listen() tests to test unix sockets, and added a (currently disabled in shadow) additional listen() test.

### CODE REVIEW

File: src/main/host/descriptor/socket/unix.rs [hunk 1]
```diff
@@ -1320,6 +1444,17 @@ impl UnixSocketCommon {
     }
 }
 
+fn backlog_to_queue_size(backlog: i32) -> u32 {
+    // linux also makes this cast, so negative backlogs wrap around to large positive backlogs
+    // https://elixir.free-electrons.com/linux/v5.11.22/source/net/unix/af_unix.c#L628
```
Comment (sporksmith) [inline src/main/host/descriptor/socket/unix.rs:1449]: Is this a bug or a feature in Linux? If it's a bug, maybe worth logging a warning when backlog is negative?

File: src/main/host/descriptor/socket/unix.rs [hunk 2]
```diff
@@ -1320,6 +1444,17 @@ impl UnixSocketCommon {
...
+    let backlog = backlog as u32;
+
+    // the linux '__sys_listen()' applies the somaxconn max to all protocols, including unix sockets
+    let queue_limit = std::cmp::min(backlog, unsafe { c::SHADOW_SOMAXCONN }.try_into().unwrap());
+
+    queue_limit.saturating_add(1)
```
Comment (sporksmith) [inline src/main/host/descriptor/socket/unix.rs:1455]: Why +1? (Probably worth a comment)

**Review by sporksmith — APPROVED** [review]

File: src/main/host/descriptor/socket/unix.rs (hunk 1 above)
Comment (stevenengler) [reply src/main/host/descriptor/socket/unix.rs:1449]: I don't think it's a bug or a feature, just a legacy artifact that probably can't be changed without breaking some userland programs. I don't think a warning is needed as it's not a bug or issue in shadow.

File: src/main/host/descriptor/socket/unix.rs (hunk 2 above)
Comment (stevenengler) [reply src/main/host/descriptor/socket/unix.rs:1455]: Just because that's what Linux does :) But I'll add a comment here.

**Pull request merged**